src/runtime-rs/crates/hypervisor/src/device/util.rs

```diff
@@ -88,4 +110,61 @@ mod tests {
             assert_eq!(&out, output);
         }
     }
+
+    #[test]
+    fn test_do_increase_count() {
+        // First, ref_count is 0
+        let ref_count_0 = &mut 0_u64;
+        let _ = do_decrease_count(ref_count_0).is_err();
+
+        assert!(!do_increase_count(ref_count_0).unwrap());
+        assert!(!do_decrease_count(ref_count_0).unwrap());
+
+        // Second, ref_count > 0
+        let ref_count_3 = &mut 3_u64;
+        assert!(do_increase_count(ref_count_3).unwrap());
+        assert!(do_decrease_count(ref_count_3).unwrap());
+
+        // Third, ref_count is MAX
+        let mut max_count = std::u64::MAX;
+        let ref_count_max: &mut u64 = &mut max_count;
+        let _ = do_increase_count(ref_count_max).is_err();
```

Comment:
no assert here?

Reply:
xxx`.is_err()` is doing the assertion work. test cast pass !